### ISSUE TITLE
Calendar: UTC/time zone support

### ISSUE THREAD

**Issue by inad9300** [primefaces/primereact#4128]
### Describe the feature you would like to see added

Right now, dates typed in a calendar's input box seem to be interpreted as local times. However, this is not always what's desirable. In my case, I would like to be able to type a UTC date and have it be interpreted as a UTC date, rather than local time.

**Comment by melloware** [comment]
Similar to: https://github.com/primefaces/primereact/issues/4076

**Comment by inad9300** [comment]
Yeah, absolutely; sorry, I looked for "timezone" and didn't fine anything related. In fact, your comment there is quite on point... I'm rather certain now that this was a logical mistake on my side. Thanks for your help, and apologies for the noise!

**Comment by melloware** [comment]
OK I am going to close that other ticket too.

**Comment by inad9300** [comment]
Wait, my bad... Damn, dates are tricky. An easy way to see the problem is: the initial time used by the calendar corresponds to the local time of the machine that opens it. More generally, the representation of the value both in the selection elements and in the input box is as local time.

The representation in the input box is in fact configurable by the prop `formatDateTime`. However, the representation in the selection elements is not and shows only local time. That's what's missing.

**Comment by melloware** [comment]
Can you check this: https://github.com/primefaces/primereact/issues/3381

I thought it allowed you to set both...

**Comment by melloware** [comment]
OK there are 11 spots in the code using `new Date()` that probably need to check the "locale" if set else you will just get the Browser date time.

**Comment by inad9300** [comment]
Note that there currently isn't a "time zone" setting in the "locale". Which I think makes sense, since there are many "locales" with more than one time zone. It probably should be a separate setting.

**Comment by inad9300** [comment]
Any progress on this front? It's a rather fundamental feature for a calendar picker, I would say.

**Comment by melloware** [comment]
I was looking it has not been implemented in PrimeVue yet either.  Maybe we have to be pioneers and be the first of the libs to implement this.

**Comment by softwarega** [comment]
[Fullcalendar](https://fullcalendar.io/docs/timeZone) has it already implemented, I think that could be a good starter point to see how they are handling the TZ feature there.

**Comment by ImmortalRabbit** [comment]
Any progress on this? It will be really helpful to be able to pass timezone to calendar.

**Comment by VR4CHeet** [comment]
Is there any update for the setting the timezone for the selecting value or it should be same timezone as the browser's timezone!?
